packages/site-kit/components/docs/Contents.svelte

```diff
@@ -0,0 +1,257 @@
+<script>
+	import { afterUpdate } from 'svelte';
+	import Icon from '../Icon.svelte';
+
+	export let contents = [];
+	export let path = null;
+	export let prevent_sidebar_scroll = false;
+
+	let show_contents = false;
+	let ul;
+
+	afterUpdate(() => {
+		// bit of a hack — prevent sidebar scrolling if
+		// TOC is open on mobile, or scroll came from within sidebar
+		if (prevent_sidebar_scroll || show_contents && window.innerWidth < 832) return;
+
+		const active = ul.querySelector('.active');
+
+		if (active) {
+			const { top, bottom } = active.getBoundingClientRect();
+
+			const min = 200;
+			const max = window.innerHeight - 200;
+
+			if (top > max) {
+				ul.parentNode.scrollBy({
+					top: top - max,
+					left: 0
+				});
+			} else if (bottom < min) {
+				ul.parentNode.scrollBy({
+					top: bottom - min,
+					left: 0
+				});
+			}
+		}
+	});
+</script>
+
+<aside class="sidebar-container" class:open={show_contents}>
+	<div class="sidebar" on:click={() => (show_contents = false)}>
+		<!-- scroll container -->
+		<ul
+			bind:this={ul}
+			class="reference-toc"
+			on:mouseenter={() => (prevent_sidebar_scroll = true)}
+			on:mouseleave={() => (prevent_sidebar_scroll = false)}
+		>
+			{#each contents as section}
+				<li>
+					<a class="section" class:active={section.path === path} href={section.path}>
+						{section.title}
+					</a>
+
+					{#each section.sections as subsection}
+						<a class="subsection" class:active={subsection.path === path} href={subsection.path}>
+							{subsection.title}
+						</a>
+
+						{#each subsection.sections as subsection}
+							<a
+								class="nested subsection"
+								class:active={subsection.path === path}
+								href={subsection.path}
+							>
+								{subsection.title}
+							</a>
+						{/each}
+					{/each}
+				</li>
+			{/each}
```

Comment:
Should we make these nested ul > li > ul for better semantics and probably some minor a11y improvements?

Reply:
probably, yeah — there might have been some reason why it wasn't done that way in the first place but will try it

Reply:
Original it was a flat list of nav elements so was trickier. When I redid the docs formatting I changed it to nested so this would be easier. We just never got round to making use of it.